server/service/async/async.go

```diff
@@ -17,94 +18,75 @@ import (
 const collectorLockKey = "locks:async_collector:{%s}"
 
 type Task struct {
-	Datastore fleet.Datastore
-	Pool      fleet.RedisPool
-	Clock     clock.Clock
-	// AsyncEnabled indicates if async processing is enabled in the
-	// configuration. Note that Pool can be nil if this is false.
-	AsyncEnabled bool
-
-	LockTimeout        time.Duration
-	LogStatsInterval   time.Duration
-	InsertBatch        int
-	DeleteBatch        int
-	UpdateBatch        int
-	RedisPopCount      int
-	RedisScanKeysCount int
-	CollectorInterval  time.Duration
-
+	datastore   fleet.Datastore
+	pool        fleet.RedisPool
+	clock       clock.Clock
+	taskConfigs map[string]config.AsyncProcessingConfig
```

Comment:
thinking out loud, not even sure if it's a good idea: I wonder if instead of a raw `string` we should declare our own custom type (which just alias to `string`) to better communicate that there's a well defined and constant set of tasks which can be configured.

kind of the classic:

```go
type TaskName string

const (
	AsyncTaskLabelMembership TaskName  = "label_membership"
	AsyncTaskPolicyMembership TaskName = "policy_membership"
	AsyncTaskHostLastSeen     TaskName = "host_last_seen"
)
```

Reply:
Yeah there's not too much value type-safety-wise because you can still pass a string constant and it will accept it fine, but I guess it's true that it communicates that there's some set defined somewhere and new ones should be added there.

Reply:
indeed, it might be a good idea just to communicate intent, but I don't think it's super important!

offtopic but tangentially related I was reading this [blog post](https://npf.io/2022/05/safer-enums/) the other day that talks about a way to enforce type safety in a simple(ish) way:

```go
type FlagID struct {
    name string
}
func (f FlagID) String() { return f.name } 

var (
    FooBar = FlagID{ “FooBar” }
    FizzBuzz = FlagID{ “FizzBuzz” }
)
```

Reply:
Yeah I saw that recently on HN I think? That's a clever approach.